source/Engine.cpp

```diff
@@ -1623,6 +1623,31 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 		// self-destruct.
 		if(ship->IsDestroyed())
 		{
+			if(!ship->CoreShip().empty())
+			{
+				const Ship *shipToPlace = GameData::Ships().Get(ship->CoreShip());
+				if(shipToPlace->IsValid())
+				{
+					const shared_ptr<Ship> escapeShip = make_shared<Ship>(*shipToPlace);
+					escapeShip->Recharge();
+					escapeShip->SetName(ship->Name());
+					escapeShip->SetGovernment(ship->GetGovernment());
+					if(ship->CorePersonality().IsValid())
```

Comment:
idk about this name, feels more this should be IsDefined()?
And I dont think we need this, cant we just have corepersonality as a pointer to a personality and then if its nullptr then dont use it?

Reply:
done